### PR TITLE
feat(grammar): improve parsing

### LINKED ISSUE #127: `rename` management
Here is a sample of a `rename` instruction:
`rename pt to ptLanguage forClass MultilingualExpression`

the structure is
`rename <elt1> to <elt2> forClasss <className>`

### LINKED ISSUE #128: check() instruction
The expressions like `(condition).check()` are not handled.

### PR DESCRIPTION
**Pull request by alacambre-yseop**
- improve the parsing of the `forall` instruction;
- accept the `(condition)check()`-like expressions.

fixes  #128  
fixes #127

### CODE REVIEW

File: server/src/test/Parsing.test.ts [hunk 1]
```diff
@@ -26,6 +26,17 @@ function checkInputValidityForRule(ruleToTest: (parser: YmlParser) => ParserRule
 }
 
 describe('Parsing Tests', () => {
+    describe('intruction_rename rule', () => {
+        it('should parse without errors an `intruction_rename` instruction', (done) => {
+            checkInputValidityForRule((parser) => parser.instruction_rename(), `rename myAttr to myOtherAttr forClass MyClass`);
+            done();
+        });
+        it('should parse without errors an `intruction_rename` instruction using other existing keywords', (done) => {
+            checkInputValidityForRule((parser) => parser.instruction_rename(), `rename rename to forClass forClass Function`);
```
Comment (davanessoff-yseop) [inline server/src/test/Parsing.test.ts]: `rename rename` ?

Comment (alacambre-yseop) [reply server/src/test/Parsing.test.ts]: This is intended `should parse without errors an `intruction_rename` instruction using other existing keywords'. Except if I can't have an attribute named `rename` in a class.

Comment (davanessoff-yseop) [reply server/src/test/Parsing.test.ts]: Not sure you can. Some words are problematic, I think. Like naming your attribute `null`. That could be tested.

Comment (alacambre-yseop) [reply server/src/test/Parsing.test.ts]: This compiles
```
rename    pt to rename forClass MultilingualExpression
rename    rename to forClass forClass MultilingualExpression
rename    forClass to ptLanguage forClass MultilingualExpression
```

Comment (alacambre-yseop) [reply server/src/test/Parsing.test.ts]: This does too:
```
rename    pt to null forClass MultilingualExpression
rename    null to forClass forClass MultilingualExpression
rename    forClass to ptLanguage forClass MultilingualExpression
```

Comment (alacambre-yseop) [reply server/src/test/Parsing.test.ts]: This compiles too:
```
rename    pt to null forClass MultilingualExpression

MultilingualExpression patate
--> null "Hello"
;
```

Comment (davanessoff-yseop) [reply server/src/test/Parsing.test.ts]: I'm convinced

**Pull request merged**